dask/blockwise.py

```diff
@@ -188,6 +188,30 @@ def __dask_distributed_unpack__(cls, state):
         return cls(**state)
 
 
+class BlockIndex(BlockwiseDep):
+    """Index BlockwiseDep argument
+
+    The purpose of this class is to provide each
+    block of a ``Blockwise``-based operation with
+    the current index.
+    """
+
+    produces_tasks: bool = False
+
+    def __init__(self, numblocks: Tuple[int, ...]):
+        self.numblocks = numblocks
```

Comment:
```suggestion
        # NOTE: unused, just needs to be set to follow the `BlockwiseDep` interface
        self.numblocks = numblocks
```
It confused me for a bit why this was necessary